src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -90,63 +104,52 @@ def __init__(
         self.architecture = arch
 
         # Qiskit uses integer indices to refer to qubits, so we need to map component names to indices.
-        qb_to_idx = {qb: idx for idx, qb in enumerate(arch.components)}
-        operations = {gate_name: gate_info.loci for gate_name, gate_info in arch.gates.items()}
-        target = Target()
-
-        def _create_properties(
-            op_name: str, symmetric: bool = False
-        ) -> dict[tuple[int, ...], InstructionProperties | None]:
-            """Creates the Qiskit instruction properties dictionary for the given IQM native operation.
-
-            Currently we do not provide any actual properties for the operation other than the
-            allowed loci.
-            """
-            loci = operations[op_name]
-            if symmetric:
-                # For symmetric gates, construct all the valid loci for Qiskit.
-                # Coupling maps in Qiskit are directed graphs, and gate symmetry is provided explicitly.
-                loci = tuple(permuted_locus for locus in loci for permuted_locus in itertools.permutations(locus))
-            return {tuple(qb_to_idx[qb] for qb in locus): None for locus in loci}
+        # Because of the way the Target and the transpiler interact, the resonators need to have higher indices than
+        # qubits, or else transpiling with optimization_level=0 will fail because of lacking resonator indices.
+        qb_to_idx = {
+            qb: idx
+            for idx, qb in enumerate(
+                sorted(arch.qubits, key=_component_sort_key)
+                + sorted(arch.computational_resonators, key=_component_sort_key)
```

Comment:
These are already sorted in the DQA.

Reply:
You sure? The sorting was already there

Reply:
There is an old unittests that checks if the qubits names are properly sorted. i.e. tests/test_iqm_backend_base.py:test_qubit_name_to_index_to_qubit_name